Sprint-3/revise/implement/password-validator.test.js

```diff
@@ -14,3 +14,66 @@ To be valid, a password must:
 
 You must breakdown this problem in order to solve it. Find one test case first and get that working
 */
+
+function isValidPassword(password, previousPasswords) {
+ 
+  if (password.length < 5) {
+    return false;
+  }
+
+  if (!/[A-Z]/.test(password)) {
+    return false;
+  }
+
+  if (!/[a-z]/.test(password)) {
+    return false;
+  }
+
+  if (!/\d/.test(password)) {
+    return false;
+  }
+
+  if (!/[!#$%.&*]/.test(password)) {
+    return false;
+  }
+
+  if (previousPasswords.includes(password)) {
+    return false;
+  }
+
+  return true;
+}
+
+
+describe('isValidPassword', () => {
+
+  const previousPasswords = ["Danial123?", "helloWorld1", "Test@123"];
+
+  test('should return false if password is less than 5 characters', () => {
+    expect(isValidPassword('1234', previousPasswords)).toBe(false);
```

Comment:
"1234" is not specific enough. "Ab$1" would be a better test value because it would pass all other checks except the check of password length.

Reply:
I've changed the password to Danial$1.